app/models/payment.rb

```diff
@@ -0,0 +1,75 @@
+class Payment < ActiveRecord::Base
+  has_many :ticket_purchases
+  belongs_to :user
+  belongs_to :conference
+
+  attr_accessor :credit_card_number
+  attr_accessor :credit_card_type
+  attr_accessor :card_verification_value
+  attr_accessor :expiration_month
+  attr_accessor :expiration_year
+
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :credit_card_number, presence: true
+  validates :card_verification_value, presence: true, length: { minimum: 3, maximum: 4 }
+  validates :expiration_month, presence: true, numericality: { greater_than_or_equal_to: 1, less_than_or_equal_to: 12 }
+  validates :expiration_year, presence: true
+  validates :amount, presence: true, numericality: { greater_than: 0 }
+
+  validate :validate_card, on: create
+
+  enum status: {
+    unpaid: 0,
+    success: 1,
+    failure: 2
+  }
+
+  def credit_card
+    @credit_card = ActiveMerchant::Billing::CreditCard.new(
+      first_name:          first_name,
+      last_name:           last_name,
+      number:              credit_card_number,
+      month:               expiration_month,
+      year:                expiration_year,
+      verification_value:  card_verification_value
+    )
+  end
+
+  def validate_card
+    unless credit_card.validate.empty?
+      credit_card.errors.full_messages.each do |message|
+        errors.add(:base, message)
```

Comment:
Why do you mess with the error collection? The validations already add errors to it? I don't get it, can you explain?